src/components/BulkEnrollmentResultsDownloadPage/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Redirect, useParams } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform';
+import { ToastsContext } from '../Toasts';
+import LicenseManagerApiService from '../../data/services/LicenseManagerAPIService';
+
+const BulkEnrollmentResultsDownloadPage = ({ enterpriseId }) => {
+  const { enterpriseSlug, bulkEnrollmentJobId } = useParams();
+  const { addToast } = useContext(ToastsContext);
+  const [isLoading, setLoading] = useState(true);
+  const [redirectUrl, setRedirectUrl] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (enterpriseId && isLoading) {
+      LicenseManagerApiService.fetchBulkEnrollmentJob(enterpriseId, bulkEnrollmentJobId)
+        .then((response) => {
+          const result = camelCaseObject(response.data);
+          if (result.downloadUrl) {
+            setRedirectUrl(result.downloadUrl);
+          } else {
+            throw new Error('Your download is not ready yet.');
+          }
+        })
+        .catch((err) => {
+          logError(err);
+          setError(err);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  });
+
+  if (isLoading) {
+    return <h1>loading...</h1>;
+  }
+  if (error) {
+    addToast('There was a problem with your request.');
```

Comment:
In the current toast message, it seems a problem with the API (e.g., a 500) is synonymous with the download simply not being ready yet? Would we want/need to distinguish between these so the user knows that "oh, i just need to give it more time. i'll check back in 5 minutes" vs. "i need to reach out to support because it says a problem keeps occurring"?